src/components/menu/submenu-block/submenu-block.component.js

```diff
@@ -1,27 +1,38 @@
-import React from "react";
+import React, { useContext } from "react";
 import PropTypes from "prop-types";
-import { StyledSubmenuBlock } from "./submenu.style";
+import StyledSubmenuBlock from "./submenu-block.style";
+import { MenuContext } from "../menu.component";
+import Logger from "../../../utils/logger/logger";
+
+let deprecatedWarnTriggered = false;
+
+const SubmenuBlock = ({ children, variant = "default" }) => {
+  const menuContext = useContext(MenuContext);
+
+  if (!deprecatedWarnTriggered) {
+    deprecatedWarnTriggered = true;
+    // eslint-disable-next-line max-len
+    Logger.deprecate(
+      "`styleOverride` that is used in the `Fieldset` component is deprecated and will soon be removed."
```

Comment:
As mentioned in private conversation, this needs updating

Reply:
Do we need this warning at all?

Reply:
The component may have been used even though it is stripped away inside `MenuItem`